### PR TITLE
Add ability to use variables in email subject

### PR DESCRIPTION
**Pull request by andie-chernysh**

### CODE REVIEW

File: lib/effective/liquid_mailer.rb [hunk 1]
```diff
@@ -2,11 +2,14 @@ module Effective
...
-      mail_method = caller_locations(1,1)[0].label
-      email_template = EffectiveEmailTemplates.get(mail_method)
-      headers = headers.merge(email_template.mail_options)
-      super(headers, &block)
+      mail_method = caller_locations(1, 1)[0].label
+      options = EffectiveEmailTemplates.get(mail_method).mail_options
+
+      if EffectiveEmailTemplates.allow_variables_in_email_subject && options[:subject].present?
+        options[:subject] = Liquid::Template.parse(options[:subject]).render(@to_liquid) rescue options[:subject]
+      end
```
Comment (andie-chernysh) [inline lib/effective/liquid_mailer.rb]: @jfeaver, I coudn't find better place to add it than here. Actually, it is the one place when both `to_liquid` data hash and email subject are crossed.

**Comment by jfeaver** [comment]
@andie-chernysh, can you use the template's `render` method here I wonder?

https://github.com/code-and-effect/effective_email_templates/blob/master/lib/effective_email_templates/email_view_template.rb#L10

Alternatively, can you do something like `self.subject = Liquid::Template.parse( subject )` here: 

https://github.com/code-and-effect/effective_email_templates/blob/master/app/models/effective/email_template.rb#L20

Then, you could render the subject in the Mailer as you have done but it would simply be:

`options[:subject] = options[:subject].render(@to_liquid) if @to_liquid.present?`

Finally, I don't think we need to make this a configuration option. The gem should always allow for variables in the subject line. Did you run into problems while implementing this feature if it wasn't configurable (or why did you decide to make it configurable)?

**Comment by andie-chernysh** [comment]
@jfeaver, I have just updated this feature to be non-configurable.

**Comment by matt-riemer** [comment]
effective_email_templates 0.3.5 pushed

**Pull request merged**